### DISCUSSION TITLE
19. HTTPS로 모든 보안 문제를 해결할 수 있나요?

### DISCUSSION THREAD

**Discussion by eun-byeol** [jewelry-box/cs-interview#21] [General]
HTTPS로 모든 보안 문제를 해결할 수 있나요?

**Comment by linirini** [comment]
HTTPS는 데이터 전송 중 보안을 강화하는 중요한 방법이지만, 모든 보안 문제를 해결할 수 있는 만능 도구는 아니다.
### HTTPS의 주요 보안 기능
- 데이터 암호화
  HTTPS는 클라이언트와 서버 간 데이터 전송을 암호화하여 중간에 탈취당하더라도 읽을 수 없도록 보호한다.
- 데이터 무결성
  전송된 데이터가 네트워크 상에서 변조되지 않도록 보장한다.
- 서버 인증
  SSL/TLS 인증서를 통해 서버가 신뢰할 수 있는 기관에서 발급받은 것임을 증명한다. 이를 통해 가짜 서버가 아님을 확인할 수 있다.
### HTTPS의 한계
- 애플리케이션 레벨의 취약점
  HTTPS는 전송 계층에서 데이터를 암호화하여 보안을 제공하지만, **애플리케이션 자체의 취약점(논리적 보안 문제)**을 보호하지는 못한다.
  ex) SQL Injection, XSS, CSRF과 같은 웹 애플리케이션 보안 공격
- SSL/TLS 인증서 관리 문제
  잘못된 인증서를 발급받거나, 인증서가 만료된 경우 보안이 취약해질 수 있다.
  중간자 공격으로 신뢰할 수 없는 서버로 접속하게 될 위험이 존재한다.
- 네트워크 및 물리적 보안
  HTTPS는 네트워크 상 데이터를 보호하지만, 네트워크 자체가 손상된 경우 보호할 수 없다.
